File: LPIS_1/src/main/java/com/example/lab/interpreter/InterpreterContext.java
package com.example.lab.interpreter;

import com.example.lab.FunctionContext;

import java.util.List;

public interface InterpreterContext extends FunctionContext {

    void returnVoid();

    void callFunctionVoid(String name, List<Object> arguments);

    void setValue(String name, int value);

    void setValue(String name, List<Integer> value);

    int getValueElement(String name);

    List<Integer> getValueList(String name);

    InterpreterContext block();

}

File: LPIS_1/src/main/java/com/example/lab/interpreter/InterpreterContext.kt
package com.example.lab.interpreter

import com.example.lab.FunctionContext

interface InterpreterContext : FunctionContext {

	fun returnVoid()

	fun callFunctionVoid(name: String, arguments: List<Any>)

	fun setValue(name: String, value: Int)

	fun setValue(name: String?, value: List<Int>)

	fun getValueElement(name: String): Int

	fun getValueList(name: String): List<Int>

	override fun block(): InterpreterContext
}
